thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/alertFilterAutotune/BaseAlertFilterAutoTune.java

```diff
@@ -0,0 +1,40 @@
+package com.linkedin.thirdeye.anomalydetection.alertFilterAutotune;
+
+import com.linkedin.thirdeye.datalayer.dto.AutotuneConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.detector.email.filter.AlertFilter;
+import com.linkedin.thirdeye.detector.email.filter.PrecisionRecallEvaluator;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public abstract class BaseAlertFilterAutoTune implements AlertFilterAutoTune {
+  private final static Logger LOG = LoggerFactory.getLogger(BaseAlertFilterAutoTune.class);
+
+  protected AutotuneConfigDTO autotuneConfig;
+  protected List<MergedAnomalyResultDTO> trainingAnomalies;
+  protected AlertFilter alertFilter;
+
+  public void init(List<MergedAnomalyResultDTO> anomalies, AlertFilter alertFilter, AutotuneConfigDTO autotuneConfig) {
```

Comment:
don't you need a constructor here? are anomalies and alertFilters independent in principle?